File: ml/lib/normalize_katalog_code.py
"""Normalisasi kode katalog gaya lama (AppSheet) -> kode katalog WARNOTO sekarang.

Dipisah dari train_forecast.py supaya bisa diimpor+ditest tanpa menarik
dependency berat (prophet, supabase) yang butuh env var/koneksi network.
"""
import re


def normalize_katalog_code(raw):
    """Ambil digit saja; kalau 10 digit & diawali '100', buang 3 digit pertama."""
    digits = re.sub(r"[^0-9]", "", str(raw))
    if len(digits) == 10 and digits.startswith("100"):
        return digits[3:]
    return digits

File: src/lib/normalizeKatalogCode.js
// Normalisasi kode katalog gaya lama (AppSheet) -> kode katalog WARNOTO sekarang.
// Port JS dari ml/lib/normalize_katalog_code.py — logic HARUS identik, jangan menyimpang.
export function normalizeKatalogCode(raw) {
  const digits = String(raw ?? "").replace(/[^0-9]/g, "");
  if (digits.length === 10 && digits.startsWith("100")) return digits.slice(3);
  return digits;
}
